resolwe_bio/processes/clustering/find_simmilar.py

```diff
@@ -0,0 +1,207 @@
+"""Find genes with similar expression profile."""
+import json
+from collections import defaultdict
+
+import pandas as pd
+from scipy.stats import pearsonr, spearmanr
+
+from resolwe.process import (
+    DataField,
+    JsonField,
+    ListField,
+    Process,
+    SchedulingClass,
+    StringField,
+)
+
+
+def get_expression(fname, sep="\t"):
+    """Read expressions from file."""
+    df = pd.read_csv(
+        filepath_or_buffer=fname,
+        sep=sep,
+        header=0,
+        index_col=0,
+        compression="gzip",
+        dtype={
+            0: str,
+            1: float,
+        },
+        keep_default_na=False,
+    )
+    df.index = df.index.map(str)
+    return df
+
+
+def get_average_expression(fnames, name, sep="\t"):
+    """Get average expression for replicates of one time point."""
+    dfs = [get_expression(fname, sep=sep) for fname in fnames]
+    joined = pd.concat(dfs, axis=1, join="outer", sort=True)
```

Comment:
I believe that we need inner join here. With outer join we'll encounter NaNs in the mean in case where the indices are not permutationally equal, i.e., where one index contains different genes than another index. Then in `join_expressions` such genes will propagate further in both inner and outer join. Write a test for this scenario.

Sorting is redundant here. It suffices that the genes are sorted by distance at the end.

Reply:
You are correct, NaNs are propagated further and using inner join is a better approach.